tests/ui/associated-type-bounds/suggest-removing-impl.rs

```diff
@@ -0,0 +1,7 @@
+trait Tr {
```

Comment:
Does this also work for generic bounds like `<T: impl Sized>`? What about other "bounds" positions? Can you flesh out tests for any more cases where this applies?

Reply:
Wait, didn't see that we already did it for generics.

Reply:
Added a test for that.